Sources/OpenSSL/OpenSSLClient.swift

```diff
@@ -15,8 +15,12 @@ public final class OpenSSLClient: TLSClient {
     private let tcp: TCPClient
 
     /// Create a new `OpenSSLClient`
-    public init(tcp: TCPClient, using settings: TLSClientSettings) throws {
-        let socket = try OpenSSLSocket(tcp: tcp.socket, method: .tls1_2, side: .client)
+    public init(
+        tcp: TCPClient,
+        using settings: TLSClientSettings,
+        configuration: OpenSSLSettings = OpenSSLSettings()
```

Comment:
the double settings thing here is quite confusing. Maybe we can come up w/ a different name? Or perhaps `TLSClientSettings` can have a generic parameter `custom: T` which stores additional custom settings?

Reply:
That's my problem, too. I don't really know how.

I'll add comments, but I'm not sure if I can easily add tests here.